migtests/tests/analyze-schema/dummy-export-dir/schema/tables/table.sql

```diff
@@ -419,3 +419,33 @@ CREATE TABLE timestamptz_multirange_table (
     global_event_times tstzmultirange
 );
 
+-- Testing tables with unique nulls not distinct constraints
+
+-- Control case
+CREATE TABLE users_unique_nulls_distinct (
+    id SERIAL PRIMARY KEY,
+    email TEXT,
+    UNIQUE (email)
+);
+
+CREATE TABLE users_unique_nulls_not_distinct (
+    id SERIAL PRIMARY KEY,
+    email TEXT,
+    UNIQUE NULLS NOT DISTINCT (email)
+);
+
+CREATE TABLE sales_unique_nulls_not_distinct (
+    store_id INT,
+    product_id INT,
+    sale_date DATE,
+    UNIQUE NULLS NOT DISTINCT (store_id, product_id, sale_date)
+);
+
+CREATE TABLE sales_unique_nulls_not_distinct_alter (
+	store_id INT,
+	product_id INT,
+	sale_date DATE
+);
+
+ALTER TABLE sales_unique_nulls_not_distinct_alter
+	ADD CONSTRAINT sales_unique_nulls_not_distinct_alter_unique UNIQUE NULLS NOT DISTINCT (store_id, product_id, sale_date);
```

Comment:
This can also come up as a `CREATE UNIQUE INDEX` not just constraints.. https://www.postgresql.org/docs/current/indexes-unique.html

I think we should check for that as well? cc: @priyanshi-yb